pathauto.routing.yml

```diff
@@ -1,11 +1,91 @@
-pathauto.patterns.form:
+entity.pathauto_pattern.collection:
   path: '/admin/config/search/path/patterns'
   defaults:
-    _form: '\Drupal\pathauto\Form\PathautoPatternsForm'
+    _entity_list: 'pathauto_pattern'
     _title: 'Patterns'
   requirements:
     _permission: 'administer pathauto'
 
+entity.pathauto_pattern.add_form:
+  path: '/admin/config/search/path/patterns/add'
+  defaults:
+    _entity_wizard: 'pathauto_pattern.add'
+    _title: 'Add Pathauto pattern'
+    tempstore_id: 'pathauto.pattern'
+  requirements:
+    _permission: 'administer pathauto'
+
+entity.pathauto_pattern.edit_form:
+  path: '/admin/config/search/path/patterns/{machine_name}/{step}'
+  defaults:
+    _entity_wizard: 'pathauto_pattern.edit'
+    _title: 'Edit Pathauto pattern'
+    tempstore_id: 'pathauto.pattern'
+  requirements:
+    _permission: 'administer pathauto'
+
+entity.pathauto_pattern.delete_form:
+  path: '/admin/config/search/path/patterns/{pathauto_pattern}/delete'
+  defaults:
+    _entity_form: 'pathauto_pattern.delete'
+    _title: 'Delete Pathauto pattern'
+  requirements:
+    _permission: 'administer pathauto'
```

Comment:
Most routes don't use entity upcasting but this could use _entity_access: pathauto_pattern.delete